### DISCUSSION TITLE
Is it possible to return values from an rmd or qmd back to the pipeline?

### DISCUSSION THREAD

**Discussion by mbrousil** [ropensci/tarchetypes#125] [Help] [answered]
I'm working with a pipeline that generates several reports with `tar_render()` (eventually `tar_quarto()`), and I'm wondering if it's possible to return values calculated within the reports back to the pipeline for use downstream?

For example (not intended to run),

`_targets.R`:
```
list(
  tar_file_read(my_data,
                "data/my_data.csv"),
  
  tar_file_read(not_my_data,
                "data/not_my_data.csv"),
  
  tar_render(cleaning_report,
             "R/cleaning_report.Rmd"),
  
  tar_target(post_clean_join,
             left_join(
               x = cleaning_report$clean_data,
               y = not_my_data)
)
```

`cleaning_report.Rmd`:
````
---
title: "Cleaning report"
output: html_document
date: "2022-11-16"
---

Illustrate cleaning the data:
```{r}
tar_load(my_data)

clean_data <- my_data %>%
  filter(!is.na(value))

ggplot(clean_data) +
  geom_point(aes(x = x, y = value))
```

<!-- Now return the cleaned data back to the pipeline -->
```{r}
return(clean_data)
```

````

I found [this](https://stackoverflow.com/questions/58315771/can-rmarkdown-return-a-value-to-a-target) from `drake` but assumed there was probably an updated method in `targets`. Thanks for your help and the great package!

**Comment by wlandau** [comment, marked as answer]
By design, `targets` does not have a good way to return data objects from an Rmd/qmd report. It is cleaner to create an ordinary target for an object like that.
